### PR TITLE
Fix #4863: Add tests

### LINKED ISSUE #4863: Prevent Pattern Match Exhaustivity Warning for `transparent`
Hi, I believe that pattern match exhaustivity warnings should not be emitted for top-level pattern match statements in `transparent` methods, given that the pattern match is checked and expanded at compile-time anyway.

I noticed these warnings are emitted when matching over a sealed trait/class in a transparent method body.

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #4863: Add tests

### CODE REVIEW

**Comment by LPTK** [comment]
Will it be possible to compose the message at compile-time using inline methods with constant folding or staging?

Also, an additional `compileTimeError` method would be useful, to avoid the roundabout `@compileTimeOnly("...") val res = ???; res` pattern.

**Comment by liufengyun** [comment]
Clarify the relation to `erased`.

File: tests/neg/i4863.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+sealed trait Nat
+case class S(n: Nat) extends Nat
+case object Z extends Nat
+
+inline def pred(n: Nat) = inline n match {
+  case S(m) => m
+  case Z =>
+    @compileTimeOnly("n cannot be Z") val res = ???
+    res
```
Comment (liufengyun) [inline tests/neg/i4863.scala]: `scala.compiletime.error("n cannot be Z")` does the job.

**Comment by liufengyun** [comment]
@LPTK Const folding will work for `scala.compiletime.error` in macros, if the parameter is `inline` -- for just inline methods, currently the compiler does not do that.

**Pull request merged**